Sources/MongoSwift/BSON/Document.swift

```diff
@@ -238,6 +255,35 @@ extension Document {
         try idDoc.merge(self)
         return idDoc
     }
+
+    /// Helper function for copying elements from some source document to a destination document while
+    /// excluding a non-zero number of keys
+    internal func copyElements(to otherDoc: inout Document, excluding keys: [String]) throws {
+        guard !keys.isEmpty else {
+            throw InternalError(message: "No keys to exclude, use 'bson_copy' instead")
+        }
+
+        let cStrings: [ContiguousArray<CChar>] = keys.map { $0.utf8CString }
+        var cPtrs: [UnsafePointer<CChar>] = []
+
+        for cString in cStrings {
+            let bufferPtr: UnsafeBufferPointer<CChar> = cString.withUnsafeBufferPointer { $0 }
+            guard let cPtr = bufferPtr.baseAddress else {
```

Comment:
Just as a curiosity: does this compile when just passing in the buffer pointers (or even the `ContiguousArray`s themselves) to `withVaList`?

Reply:
No it won't because `withVaList` expects an array of [`CVarArg`](https://developer.apple.com/documentation/swift/cvararg)s and neither `ContiguousArray` nor `UnsafeBufferPointer` are conforming types.